src/operations/booleans/trees/Node.js

```diff
@@ -89,28 +89,30 @@ Node.prototype = {
     } while (node !== undefined)
   },
 
-  addPolygonTreeNodes: function (polygontreenodes) {
-    let current = { 'node': this, 'polygontreenodes': polygontreenodes }
-    let node
+  addPolygonTreeNodes: function (newpolygontreenodes) {
+    let current = { 'node': this, 'polygontreenodes': newpolygontreenodes }
     let stack = []
     do {
-      node = current.node
-      polygontreenodes = current.polygontreenodes
+      let node = current.node
+      let polygontreenodes = current.polygontreenodes
 
       if (polygontreenodes.length === 0) {
         current = stack.pop()
         continue
       }
-      let _this = node
       if (!node.plane) {
-        let bestplane = polygontreenodes[0].getPolygon().plane
+        let index = 0 // default
+        index = Math.floor(polygontreenodes.length / 2)
```

Comment:
can you explain a bit ? not sure I understand the logic, thanks

Reply:
I'm not sure 100% about this logic. But I wanted to clean up some of the poor choice of variable names, and use of _this. The code is now easier to understand.

Reply:
I agreed with the simplification/ removal of **_this**, but this part : 
```index = Math.floor(polygontreenodes.length / 2)``` seems very different: why the use of Math.floor ? 
In the existing code it was always accessing at index 0 ?

Reply:
ah very nice !